gui/src/components/DirectoryBrowser.svelte

```diff
@@ -0,0 +1,124 @@
+<script lang="ts">
+  import type { ReadDirResult } from '../lib/api';
+  import HomeSVG from './mono/HomeSVG.svelte';
+  import LeftUpSVG from './mono/LeftUpSVG.svelte';
+  import Textbox from './Textbox.svelte';
+
+  export let dir: ReadDirResult;
+  export let homePath: string;
+  export let handleClick: (path: string) => void;
```

Comment:
Instead of passing functions in as properties like this, you could use an event dispatcher. See the tutorial: https://svelte.dev/tutorial/component-events

However, this seems strange in general. Why are you passing a function or using events instead of using binding? Rather than `handleClick={setWorkingDirectory}`, you would do `bind:path={workingDirectory}` from the consuming component. Then inside this component, you could do `path = whatever`

Reply:
This is because `setWorkingDirectory` also updates the promise, but yeah, I want to redo mostly the whole page's promise handling as noted above.